src/formatDate.js

```diff
@@ -8,7 +8,68 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let newFormat = '';
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  let year = '';
+  let month = '';
+  let day = '';
+
+  let count = 0;
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YY':
+        year = date.slice(count, count + fromFormat[i].length);
+
+        if (+year < 30) {
+          year = 20 + year;
+        } else {
+          year = 19 + year;
+        }
+        break;
+
+      case 'YYYY':
+        year = date.slice(count, count + fromFormat[i].length);
+        break;
+
+      case 'DD':
+        day = date.slice(count, count + fromFormat[i].length);
+        break;
+
+      case 'MM':
+        month = date.slice(count, count + fromFormat[i].length);
+        break;
+    }
+    count += fromFormat[i].length + fromSeparator.length;
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i]) {
+      case 'YY':
+        newFormat += year.slice(2);
```

Comment:
When slicing the `year` to get the last two digits, you should ensure that the `year` is in the `YYYY` format. If `year` is already in `YY` format, slicing it will result in an empty string. You may need to check the length of `year` before slicing or ensure that `year` is always in the `YYYY` format when assigned.